great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py

```diff
@@ -0,0 +1,47 @@
+from typing import List, Optional
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.rule_based_profiler.domain_builder.domain import Domain
+from great_expectations.rule_based_profiler.domain_builder.domain_builder import (
+    DomainBuilder,
+)
+from great_expectations.validator.validation_graph import MetricConfiguration
+from great_expectations.validator.validator import Validator
+
+
+class ColumnDomainBuilder(DomainBuilder):
+    def _get_domains(
+        self,
+        *,
+        validator: Optional[Validator] = None,
+        batch_ids: Optional[List[str]] = None,
+    ) -> List[Domain]:
+        """
+        Obtains and returns domains for all columns of a table.
+        """
+        if validator is None:
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"{self.__class__.__name__} requires a reference to an instance of the Validator class."
+            )
+
+        table_column_names: List[str] = validator.get_metric(
+            metric=MetricConfiguration(
+                metric_name="table.columns",
+                metric_domain_kwargs={},
```

Comment:
If we're going to return active_batch_id in domain_kwargs, we should explicitly specify it in the MetricConfiguration.

Reply:
@jcampbell Done -- in next commit to our feature branch (and will notify you when ready for re-review of this PR).  Thanks!